base/inc/AdePT/BVHNavigator.h

```diff
@@ -31,6 +33,56 @@ class BVHNavigator {
 
   static constexpr Precision kBoundaryPush = 10 * vecgeom::kTolerance;
 
+  __device__ static inline vecgeom::Daughter GetPlacedVolume(int aLVIndex, int index)
+  {
+    return vecgeom::globaldevicegeomdata::gDeviceLogicalVolumes[aLVIndex].GetDaughters()[index];
+  }
+
+  __host__ __device__ static inline vecgeom::VPlacedVolume *GetPlacedVolume(int global_index)
+  {
+#ifdef VECCORE_CUDA_DEVICE_COMPILATION
+    return &vecgeom::globaldevicegeomdata::gCompactPlacedVolBuffer[global_index];
+#else
+    return vecgeom::GeoManager::Instance().FindPlacedVolume(global_index);
```

Comment:
See my comment about using direct access to the volume by id in https://gitlab.cern.ch/VecGeom/VecGeom/-/merge_requests/999#note_7095384